examples/vector_databases/azuresearch/Getting_started_with_azure_ai_search_and_openai.ipynb

```diff
@@ -388,19 +462,33 @@
     }
    ],
    "source": [
-    "# Convert the 'id' and 'vector_id' columns to string so one of them can serve as our key field  \n",
-    "article_df['id'] = article_df['id'].astype(str)  \n",
-    "article_df['vector_id'] = article_df['vector_id'].astype(str)  \n",
-    "  \n",
-    "# Convert the DataFrame to a list of dictionaries  \n",
-    "documents = article_df.to_dict(orient='records')  \n",
-    "  \n",
-    "# Use SearchIndexingBufferedSender to upload the documents in batches optimized for indexing \n",
-    "with SearchIndexingBufferedSender(search_service_endpoint, index_name, AzureKeyCredential(search_service_api_key)) as batch_client:  \n",
-    "    # Add upload actions for all documents  \n",
-    "    batch_client.upload_documents(documents=documents)  \n",
-    "  \n",
-    "print(f\"Uploaded {len(documents)} documents in total\")  "
+    "from azure.core.exceptions import HttpResponseError\n",
+    "\n",
+    "# Convert the 'id' and 'vector_id' columns to string so one of them can serve as our key field\n",
+    "article_df[\"id\"] = article_df[\"id\"].astype(str)\n",
+    "article_df[\"vector_id\"] = article_df[\"vector_id\"].astype(str)\n",
+    "# Convert the DataFrame to a list of dictionaries\n",
+    "documents = article_df.to_dict(orient=\"records\")\n",
+    "\n",
+    "# Create a SearchIndexingBufferedSender\n",
+    "batch_client = SearchIndexingBufferedSender(\n",
+    "    search_service_endpoint, index_name, credential\n",
+    ")\n",
+    "\n",
+    "try:\n",
+    "    # Add upload actions for all documents\n",
+    "    for doc in documents:\n",
+    "        batch_client.upload_documents(documents=[doc])\n",
```

Comment:
Doesn't `upload_documents()` take an array already? Could you pass `documents` directly and in a single call instead of iterating over it and making a single-element array for each doc?

Reply:
Correct. Updated.